packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -443,6 +518,50 @@ export default function(
     }
   }
 
+  function callGetDerivedStateFromProps(workInProgress, instance, props) {
+    const {type} = workInProgress;
+
+    if (typeof type.getDerivedStateFromProps === 'function') {
+      if (__DEV__) {
+        if (
+          typeof instance.componentWillReceiveProps === 'function' ||
+          typeof instance.UNSAFE_componentWillReceiveProps === 'function'
+        ) {
+          const componentName = getComponentName(workInProgress) || 'Unknown';
+          if (!didWarnAboutWillReceivePropsAndDerivedState[componentName]) {
+            warning(
+              false,
+              '%s: Defines both componentWillReceiveProps() and static ' +
```

Comment:
Should we flat out refuse to support both in one component? Why not?

Reply:
It could be argued that we refuse to support both. I lean slightly towards warning. I don't have a strong argument.

Reply:
I just think it's easier than thinking about edge cases from using both. I might be wrong though.